sys/mips/cheri/cheri_debug.c

```diff
@@ -46,8 +46,56 @@
 #include <machine/pcb.h>
 #include <machine/sysarch.h>
 
+#include <vm/vm.h>
+#include <vm/vm_param.h>
+#include <vm/vm_extern.h>
+#include <vm/vm_page.h>
+#include <vm/vm_pageout.h>
+#include <vm/vm_map.h>
+
 #ifdef DDB
 
+static pid_t
+db_get_cap_owner(struct thread *td, void * __capability c)
+{
+	vm_map_t map;
+	vm_map_entry_t entry;
+	vm_offset_t addr;
+	boolean_t found;
+	pid_t pid;
+
+	if (td == NULL)
+		return (-1);
+
+	addr = __builtin_cheri_address_get(c);
```

Comment:
You'll want the base not the address, and to return -1 for null-derived things (could also do untagged, but may be useful to see the owner for something that looks like a capability but lost its tag)